dm/pkg/parser/common.go

```diff
@@ -72,6 +72,9 @@ type tableNameExtractor struct {
 }
 
 func (tne *tableNameExtractor) Enter(in ast.Node) (ast.Node, bool) {
+	if _, ok := in.(*ast.ReferenceDef); ok {
```

Comment:
I add two "create table" statements in the integration test. For previous versions, `create table dm_syncer_do_db_5 like dm_syncer_do_db_4;` would fail since `dm_syncer_do_db_4` is skipped. In the fixed version, this error should not occur.